### DISCUSSION TITLE
Data source: locations with OpenStreetMap (OSM)

### DISCUSSION THREAD

**Discussion by raphodn** [openfoodfacts/open-prices#335] [General]
## Currently

We currently link each Price to a Location.

### Backend

The data is fetched in the backend from [Nominatim](https://nominatim.openstreetmap.org/ui/search.html).

It is defined by a unique set of `osm_id` & `osm_type`.

### Frontend

There is currently 2 ways to find a location (OSM POI)
- full-text search using [Photon Komoot](https://photon.komoot.io/)
- ID search using Nominatim

## Idea

We're looking at returning POIs given GPS coordinates (latitude & longitude). Nominatim and Photon Komoot both do NOT support this. So we'll most likely use [Overpass](https://overpass-api.de/).

Is there a unique location search engine that would allow us to search on all 3 ?
- by keyworkds/adress
- by OSM ID
- by GPS coordinates (return all shops in a radius of X meters)

## Linked issues

- #13
- https://github.com/openfoodfacts/open-prices-frontend/issues/61
- https://github.com/openfoodfacts/open-prices-frontend/issues/37

**Comment by CloCkWeRX** [comment]
https://alltheplaces.xyz/ also features as many supermarkets as can be crawled, which goes above and beyond the OSM listings, publishing addresses, phone, opening hours, etc.

Linked via wikidata identifiers:
https://alltheplaces.xyz/wikidata.html

While the intent is to try and get these into OSM long term, this may be a useful supplementary dataset.
